### PR TITLE
fix: shoutout request body incomplete

### PR DESCRIPTION
**Pull request by chungwwei**
image

### CODE REVIEW

**Review by kevmo314 — COMMENTED** [review]
Don't we want the broadcaster user id? The moderator is just the one who sends the command, right? Or am I misunderstanding

**Comment by chungwwei** [comment]
> Don't we want the broadcaster user id? The moderator is just the one who sends the command, right? Or am I misunderstanding


the doc says `moderator_user_id` is required in the request body to get the shoutout event, and the `moderator_user_id` can be the ID of the broadcaster or a moderator. but `broadcaster_user_id` it not required. I don't know if its necessary to do
```
{
    broadcaster_user_id: twitchUserId,
    moderator_user_id: twitchUserId               <------- required in the request body           
}
```
or maybe `{ moderator_user_id: twitchUserId } ` is sufficient?

but `{ broadcaster_user_id: twitchUserId }` didn't get me any events, currently in deployed.

**Comment by kevmo314** [comment]
> > Don't we want the broadcaster user id? The moderator is just the one who sends the command, right? Or am I misunderstanding
> 
> the doc says `moderator_user_id` is required in the request body to get the shoutout event, and the `moderator_user_id` can be the ID of the broadcaster or a moderator. but `broadcaster_user_id` it not required. I don't know if its necessary to do
> 
> ```
> {
>     broadcaster_user_id: twitchUserId,
>     moderator_user_id: twitchUserId               <------- required in the request body           
> }
> ```
> 
> or maybe `{ moderator_user_id: twitchUserId } ` is sufficient?
> 
> but `{ broadcaster_user_id: twitchUserId }` didn't get me any events, currently in deployed.

I see, ok. You're right it seems like `{ moderator_user_id: twitchUserId } ` is sufficient. I think this only applies to `channel.shoutout.receive`, not `channel.shoutout.create` so the PR should be adjusted for only receive. See the first row in the docs. The docs are pretty poorly written. 😅

File: functions/src/eventsub.ts [hunk 1]
```diff
@@ -41,10 +41,15 @@ enum EventsubType {
...
-    type == "channel.raid"
-      ? { to_broadcaster_user_id: twitchUserId }
-      : { broadcaster_user_id: twitchUserId };
+  var condition: any = { broadcaster_user_id: twitchUserId };
+  if (type === "channel.raid") {
+    condition = { to_broadcaster_user_id: twitchUserId };
+  } else if (
+    type === "channel.shoutout.create" ||
+    type === "channel.shoutout.receive"
+  ) {
```
Comment (kevmo314) [inline functions/src/eventsub.ts:50]:
```suggestion
  } else if (type === "channel.shoutout.receive") {
```

**Comment by kevmo314** [comment]
Is there separate documentation for `channel.shoutout.create`?

**Comment by chungwwei** [comment]
> Is there separate documentation for `channel.shoutout.create`?

https://dev.twitch.tv/docs/eventsub/eventsub-subscription-types/#channelshoutoutcreate

I look today, they changed the requirements, now both the fields are required in `channel.shoutout.create` and `channel.shoutout.receive`

{
    broadcaster_user_id: twitchUserId,           <------- required in the request body 
    moderator_user_id: twitchUserId               <------- required in the request body           
}

**Review by kevmo314 — APPROVED** [review]

**Pull request merged**